presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -519,7 +520,7 @@ public PlanOptimizers(
                                 .add(new PushLimitIntoTableScan(metadata))
                                 .add(new PushPredicateIntoTableScan(metadata, typeAnalyzer))
                                 .add(new PushSampleIntoTableScan(metadata))
-                                .add(new PushAggregationIntoTableScan(metadata))
+                                .add(new PushAggregationIntoTableScanWithoutGroupingSets(metadata))
```

Comment:
What is the reason to discern two `PushAggregationIntoTableScan` rules?
Ie we certainly need 2 rules, as we match different patterns, but they still could be
added to optimizers flow as one entity.
See eg
https://github.com/prestosql/presto/blob/84d3b06c56f0c7083fbb6a8de0724c4ea832fd77/presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java#L488

Reply:
In this case it seemed more natural to add the withoutGroupingSet rule to be in the same set with all the other pushToTableScan rule but I can not add the withGroupingSet rule in the same set as it needs some other rules to execute before the plan takes the shape that it would match the pattern.